weather_man.py

```diff
@@ -0,0 +1,142 @@
+        [float]: [if we get numeric data]
+        [string]: [if we get string data]
+    """
+    if value == "":
+        return 0
+    else:
+        try:
+            return float(value)
+        except ValueError as ve:
+            return value
+
+
+def populate_readings_data_structure(single_day_reading):
+    """receives a single line from reading file and generates a dictionary of values
+
+    Args:
+        split_line ([String]): [A single line read from file]
+
+    Returns:
+        [Dictionary]: [A dictionary containing datapoints for a tuple]
+    """
+    single_day_reading['Max TemperatureC'] = to_correct_type(
+        single_day_reading['Max TemperatureC'])
+    single_day_reading['Mean TemperatureC'] = to_correct_type(
+        single_day_reading['Mean TemperatureC'])
+    single_day_reading['Min TemperatureC'] = to_correct_type(
+        single_day_reading['Min TemperatureC'])
+    single_day_reading['Dew PointC'] = to_correct_type(
+        single_day_reading['Dew PointC'])
+    single_day_reading['MeanDew PointC'] = to_correct_type(
+        single_day_reading['MeanDew PointC'])
+    single_day_reading['Min DewpointC'] = to_correct_type(
+        single_day_reading['Min DewpointC'])
+    single_day_reading['Max Humidity'] = to_correct_type(
+        single_day_reading['Max Humidity'])
+    single_day_reading['Mean Humidity'] = to_correct_type(
+        single_day_reading[' Mean Humidity'])
+    single_day_reading['Min Humidity'] = to_correct_type(
+        single_day_reading[' Min Humidity'])
+    single_day_reading['CloudCover'] = to_correct_type(
+        single_day_reading[' CloudCover'])
+    single_day_reading['WindDirDegrees'] = to_correct_type(
+        single_day_reading['WindDirDegrees'])
+    single_day_reading['Max VisibilityKm'] = to_correct_type(
+        single_day_reading[' Max VisibilityKm'])
+    single_day_reading['Mean VisibilityKm'] = to_correct_type(
+        single_day_reading[' Mean VisibilityKm'])
+    single_day_reading['Min VisibilitykM'] = to_correct_type(
+        single_day_reading[' Min VisibilitykM'])
+    single_day_reading['Max Wind SpeedKm/h'] = to_correct_type(
+        single_day_reading[' Max Wind SpeedKm/h'])
+    single_day_reading['Mean Wind SpeedKm/h'] = to_correct_type(
+        single_day_reading[' Mean Wind SpeedKm/h'])
+    single_day_reading['Max Gust SpeedKm/h'] = to_correct_type(
+        single_day_reading[' Max Gust SpeedKm/h'])
+    single_day_reading['Precipitationmm'] = to_correct_type(
+        single_day_reading['Precipitationmm'])
+    single_day_reading['Max Sea Level PressurehPa'] = to_correct_type(
+        single_day_reading[' Max Sea Level PressurehPa'])
+    single_day_reading['Mean Sea Level PressurehPa'] = to_correct_type(
+        single_day_reading[' Mean Sea Level PressurehPa'])
+    single_day_reading['Min Sea Level PressurehPa'] = to_correct_type(
+        single_day_reading[' Min Sea Level PressurehPa'])
+    return single_day_reading
```

Comment:
Are we using all these keys? And what if we don't convert these keys? Have you seen how python works on list of strings for such comparisons?